### PR TITLE
Major Refactoring for AST and Parser

### PR DESCRIPTION
**Pull request by OmniaGM**
- [x] Make DataManipulation and Cql3Statement sealed trait
- [x] Make DataDefinition sealed trait
- [x] Remove any un-used import
- [x] Add DDL package 
- [x] Make parseQuery, parseSchema and parseSchemaAs common to re-use them

### CODE REVIEW

File: cql-ast/src/main/scala/troy/cql/ast/Statements.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package troy.cql.ast
+
+import troy.cql.ast.dml._
+import troy.cql.ast.ddl._
+
+sealed trait DataDefinition
+sealed trait DataManipulation
```
Comment (tabdulradi) [inline cql-ast/src/main/scala/troy/cql/ast/Statements.scala]: Can you please change DataDefinition and DataManipulation to extend Cql3Statement?

File: cql-ast/src/main/scala/troy/cql/ast/Statements.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+sealed trait Cql3Statement
+
+case class DeleteStatement(
```
Comment (tabdulradi) [inline cql-ast/src/main/scala/troy/cql/ast/Statements.scala]: Also please add `final` mod on all case classes

**Comment by tabdulradi** [comment]
LGTM, only 2 small fixes mentioned above.

**Comment by OmniaGM** [comment]
@tabdulradi Check the changes now.

**Pull request merged**